### DISCUSSION TITLE
#TuYa TS0601_din

### DISCUSSION THREAD

**Discussion by anduliu** [Koenkk/zigbee2mqtt#10065] [Devices]
Hello.
Although I see that #TuYa TS0601_din is compatible with zigbee2mqtt, I cannot make a correct pairing.
The errors it gives me:
Warning Message received from device not compatible with Zigbee model 'TS0601' and manufacturer name '_TZE200_byzdayie'
Warning See: https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html.
Although I have been reading about it, I am not sure what to do. Can somebody help me
Thanks

**Comment by anduliu** [comment]
I have created a TS0601.js file with the following content:

- I have created a TS0601.js file with the following content:
const fz = require ('zigbee-herdsman-converters / converters / fromZigbee');
const tz = require ('zigbee-herdsman-converters / converters / toZigbee');
const exposes = require ('zigbee-herdsman-converters / lib / exposes');
const reporting = require ('zigbee-herdsman-converters / lib / reporting');
const e = exposes.presets;
const ea = exposes.access;

const device = {
    fingerprint: [{modelID: 'TS0601', manufacturerName: '_TZE200_byzdayie'}],
    model: 'TS0601',
    vendor: 'TuYa',
    description: 'Zigbee Tuya smart energy meter DDS238-2 Zigbee',
    fromZigbee: [
        //fz.ignore_basic_report, // Add this if you are getting no converter for 'genBasic'
        fz.tuya_data_point_dump, // This is a debug converter, it will be described in the next part
    ],
    toZigbee: [
        tz.tuya_data_point_test, // Another debug converter
    ],
    goal: {
        configureKey: 1,
    },
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint (1);
        await bind (endpoint, coordinatorEndpoint, ['genBasic']);
    },
    exposes: [
        // Here you should put all functionality that your device exposes
    ],
};
module.exports = device;

I have added in the configuration:
external_converters:
  - TS0601.js
log_level: debug
but the log gives me an error:

Zigbee2MQTT:error 2021-12-07 17:39:28: Failed to configure 'Medidor ', attempt 2 (ReferenceError: bind is not defined
    at Object.configure (/app/lib/util/externally-loaded.js:25:9)
    at Configure.configure (/app/lib/extension/configure.js:131:45)
    at Configure.onZigbeeEvent (/app/lib/extension/configure.js:113:18)
    at Controller.callExtensionMethod (/app/lib/controller.js:380:44))
/app/node_modules/zigbee-herdsman-converters/converters/fromZigbee.js:3101
                if (err) throw err;
                         ^
[Error: ENOENT: no such file or directory, open 'data/tuya.dump.txt'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'data/tuya.dump.txt'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.18.1 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.18.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-12-07T16_39_29_164Z-debug.log
[17:39:30] INFO: Handing over control to Zigbee2mqtt Core ...
> zigbee2mqtt@1.18.1 start /app
> node index.js

**Reply by anduliu** [reply]
I realized that the plugin needs to be updated with a new web address. And me breaking my head

**Reply by francescros** [reply]
Got same device and issue - how did you solved it? It is paired and working now? Thanks!

**Reply by anduliu** [reply]
If I have the same device and it is working correctly. To update in home assistant you have to put the correct zigbee2mqtt repository address in the plugin store.

**Reply by francescros** [reply]
got it working! Thanks

**Reply by gericho** [reply]
> If I have the same device and it is working correctly. To update in home assistant you have to put the correct zigbee2mqtt repository address in the plugin store.

May I ask to be more specific on how to update the repository, I'm not able to make this IDN rail meter work. Thanks!

**Reply by gericho** [reply]
> If I have the same device and it is working correctly. To update in home assistant you have to put the correct zigbee2mqtt repository address in the plugin store.

just checked the repo is ok, seems there's the need of a new converter

**Reply by gericho** [reply]
> got it working! Thanks

i checked the repo and it is OK, i can update the addon regularly so..., I have another one bought 2 months ago that works perfectly, this one seems identical but a "Ver.IVAP" is writtent on the bottom right corner, and it does not pair, unsupported device.

**Comment by anduliu** [comment]
In the repository store of your home assistant you add a new repository with the updated address https://github.com/zigbee2mqtt/hassio-zigbee2mqtt
I hope it helps you
***@***.***


Enviado desde Correo<https://go.microsoft.com/fwlink/?LinkId=550986> para Windows

De: ***@***.***>
Enviado: viernes, 2 de septiembre de 2022 19:45
Para: ***@***.***>
CC: ***@***.***>; ***@***.***>
Asunto: Re: [Koenkk/zigbee2mqtt] #TuYa TS0601_din (Discussion #10065)


If I have the same device and it is working correctly. To update in home assistant you have to put the correct zigbee2mqtt repository address in the plugin store.

just checked the repo is ok, seems there's the need of a new converter

**Comment by anduliu** [comment]
Then my friend must ask another person with experience. My experience is only based on searching and searching the web, but I have no idea what can happen to it. It's funny if you have another one working. Bring it to the coordinator for the pairing maybe that's it. If it is not, I recommend you read the documentation on the zigbee2mqtt page. I am sorry
Enviado desde Correo<https://go.microsoft.com/fwlink/?LinkId=550986> para Windows

De: ***@***.***>
Enviado: viernes, 2 de septiembre de 2022 20:00
Para: ***@***.***>
CC: ***@***.***>; ***@***.***>
Asunto: Re: [Koenkk/zigbee2mqtt] #TuYa TS0601_din (Discussion #10065)


got it working! Thanks

i checked the repo and it is OK, i can update the addon regularly so..., I have another one bought 2 months ago that works perfectly, this one seems identical but a "Ver.IVAP" is writtent on the bottom right corner, and it does not pair, unsupported device.

**Reply by xmatro** [reply]
Interesting. Because after unpairing this device didn't want to pairing again.
Support status - Not supported

And log here:
2022-09-06 22:09:11Starting interview of '0xb4e3f9fffe95a170'
Info 2022-09-06 22:09:11MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xb4e3f9fffe95a170","ieee_address":"0xb4e3f9fffe95a170","status":"started"},"type":"device_interview"}'
Info 2022-09-06 22:09:11MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_started","meta":{"friendly_name":"0xb4e3f9fffe95a170"},"type":"pairing"}'
Info 2022-09-06 22:09:16MQTT publish: topic 'zigbee2mqtt/0xb4e3f9fffe95a170', payload '{"last_seen":"2022-09-06T19:09:17.076Z"}'
Info 2022-09-06 22:09:20MQTT publish: topic 'zigbee2mqtt/0xb4e3f9fffe95a170', payload '{"last_seen":"2022-09-06T19:09:21.283Z"}'
Info 2022-09-06 22:09:26MQTT publish: topic 'zigbee2mqtt/0xb4e3f9fffe95a170', payload '{"last_seen":"2022-09-06T19:09:26.757Z"}'
Info 2022-09-06 22:09:30MQTT publish: topic 'zigbee2mqtt/0xb4e3f9fffe95a170', payload '{"last_seen":"2022-09-06T19:09:31.242Z"}'
Info 2022-09-06 22:09:36MQTT publish: topic 'zigbee2mqtt/0xb4e3f9fffe95a170', payload '{"last_seen":"2022-09-06T19:09:37.140Z"}'
2022-09-06 22:12:12MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0xb4e3f9fffe95a170","ieee_address":"0xb4e3f9fffe95a170","status":"failed"},"type":"device_interview"}'
Info 2022-09-06 22:12:12MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_failed","meta":{"friendly_name":"0xb4e3f9fffe95a170"},"type":"pairing"}'
